src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,158 @@
+module Cardano.Wallet.BlockSyncer
+  (
+    blockSyncerBracket
+  , runBlockLayerMockup
+  , BlocksConsumed(..)
+  ) where
+
+import Prelude
+
+
+import Control.Concurrent
+    ( forkIO, killThread, myThreadId, threadDelay )
+import Control.Exception
+    ( SomeException, bracket, catch )
+import Data.IORef
+    ( IORef, newIORef, readIORef, writeIORef )
+import Data.List
+    ( delete, (!!) )
+import Data.Text.Internal
+    ( showText )
+import Data.Word
+    ( Word64 )
+import Formatting
+    ( sformat, (%) )
+import qualified Formatting as F
+
+
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader (..), Hash (..) )
+
+
+
+newtype BlocksConsumed = BlocksConsumed [(Hash "BlockHeader")] deriving (Show, Eq)
+
+data BlockLayerMockup = BlockLayerMockup {
+      getBlock      :: (Hash "BlockHeader") -> IO Block
+    , getEpoch      :: Word64 -> IO [Block]
+    , getNetworkTip :: IO ((Hash "BlockHeader"),BlockHeader)
+    }
+
+runBlockLayerMockup
+    :: Int
+    -> [((Hash "BlockHeader"),Block)]
+    -> Int
+    -> IO BlockLayerMockup
+runBlockLayerMockup period blocks killTicks= do
+    initialHeaderData <- newIORef 0
+    _ <- forkIO $ countDown initialHeaderData
+    pure BlockLayerMockup {
+          getBlock      = _getBlock blocks
+        , getEpoch      = _getEpoch blocks
+        , getNetworkTip = _getNetworkTip blocks initialHeaderData
+        }
+    where
+        countDown
+            :: IORef Int
+            -> IO ()
+        countDown counter = do
+            previousCounter <- readIORef counter
+            if previousCounter >= killTicks then do
+                threadID <- myThreadId
+                killThread threadID
+            else do
+                threadDelay (period*1000*1000)
+                writeIORef counter $ min (length blocks - 1) (previousCounter + 1)
+                countDown counter
+
+
+_getBlock
+    :: [((Hash "BlockHeader"),Block)]
+    -> (Hash "BlockHeader")
+    -> IO Block
+_getBlock blocks headerHash =
+    case filter (\(h,_) -> h == headerHash) blocks of
+        [(_,block)] -> return block
+        _ -> error "Something wrong with the test blocks"
+
+_getEpoch
+    :: [((Hash "BlockHeader"),Block)]
+    -> Word64
+    -> IO [Block]
+_getEpoch blocks epoch = do
+    let bs =  filter (\(_,Block (BlockHeader theEpoch _ _) _) -> theEpoch == epoch) blocks
+    pure $ map snd bs
+
+_getNetworkTip
+    :: [((Hash "BlockHeader"),Block)]
+    -> IORef Int
+    -> IO ((Hash "BlockHeader"), BlockHeader)
+_getNetworkTip blocks currentPosition = do
+    index <- readIORef currentPosition
+    let (headerHash, Block blockHeader _ ) = (reverse blocks) !! index
+    pure (headerHash,blockHeader)
+
+blockSyncerBracket
+    :: IORef BlocksConsumed
+    -> BlockLayerMockup
+    -> Int
+    -> IO ()
+blockSyncerBracket blocksPersisted blockLayer period =
+    bracket
+      (return ()) --(putStrLn "setting up for ticking function")
+      (\_->return ()) --(\_ -> putStrLn "cleaning for ticking function")
```

Comment:
Commented-out code shouldn't be pushed / committed (unless it serves as an example for documentation), which isn't the case here :+1:

Reply:
I put it like that for the sake of future logging.

Reply:
Don't :) 
If it's not needed, don't do it. We favor simple code and refactor when we need to add functionalities :+1: